src/app/components/popup-host/popup-host.component.ts

```diff
@@ -0,0 +1,30 @@
+import {Component, Input, TemplateRef, ViewChild, ViewContainerRef} from '@angular/core';
+
+@Component({
+    selector: 'app-popup-host',
+    templateUrl: './popup-host.component.html',
+    styleUrls: ['./popup-host.component.css'],
+})
+export class PopupHostComponent {
+    isOpen = false;
+
+    @Input() set template(template: TemplateRef<unknown> | null) {
+        this.getPopup(template);
+    }
+
+    @ViewChild('popup', {read: ViewContainerRef, static: true})
+    private readonly popupViewContainer: ViewContainerRef | undefined;
+
+    private getPopup(template: TemplateRef<unknown> | null) {
+        this.popupViewContainer?.clear();
+
+        if (!template) {
+            this.isOpen = false;
+
+            return;
+        }
+
+        this.isOpen = true;
```

Comment:
Рекомендую в будущем выносить общие части:
```suggestion
        this.isOpen = !!template;

        if (!template) {
            return;
        }
```